### ISSUE TITLE
Document infrastructure-ready, control-plane-ready annotations

### ISSUE THREAD

**Issue by ncdc** [kubernetes-sigs/cluster-api-provider-aws#871]
/kind documentation

**Describe the solution you'd like**
Once #779 is merged to master, we'll need to document the 2 new annotations it added:

- `aws.cluster.sigs.k8s.io/infrastructure-ready`
- `aws.cluster.sigs.k8s.io/control-plane-ready`

We'll need to cover 2 scenarios:

1. **Upgrade**: I have a cluster that I created with CAPA before v0.3.4 (the first release that expects the annotations) (action item: manually annotate your cluster with the 2 new annotations)
2. **External control plane**: I’m using CAPA with some other control plane (manually annotate your cluster with the 2 new annotations)

**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]

**Environment:**

- Cluster-api-provider-aws version: 
- Kubernetes version: (use `kubectl version`): 
- OS (e.g. from `/etc/os-release`):

**Comment by ncdc** [comment]
These are no longer relevant in v1alpha2 and newer
/close
